src/live_restore/live_restore_fs.c

```diff
@@ -1657,6 +1657,48 @@ __validate_live_restore_path(WT_FILE_SYSTEM *fs, WT_SESSION_IMPL *session, const
     return (fh->close(fh, (WT_SESSION *)session));
 }
 
+/*
+ * __live_restore_copy_log_file --
+ *     Copy log across during recovery.
+ */
+static int
+__live_restore_copy_log_file(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+    WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh = (WTI_LIVE_RESTORE_FILE_HANDLE *)fh;
+    char *buf = NULL;
+
+    WT_RET(
+      __wt_calloc(session, 1, ((WTI_LIVE_RESTORE_FS *)S2C(session)->file_system)->read_size, &buf));
+
+    /*
+     * Break the copy into small chunks. Split the file into n chunks: the first n - 1 chunks will
+     * read a full WT_LIVE_RESTORE_READ_SIZE buffer, and the last chunk reads the remaining data.
+     */
+    size_t len = WT_MIN(lr_fh->source_size, lr_fh->destination.back_pointer->read_size);
+    for (size_t off = 0; off < lr_fh->source_size; off += len) {
+        WT_ERR(lr_fh->source->fh_read(lr_fh->source, wt_session, (wt_off_t)off, len, buf));
```

Comment:
Len needs to be computed on each iteration. Read size will change at the end of the file.